ksqldb-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceAsProperties.java

```diff
@@ -208,14 +177,26 @@ public Map<String, String> getValueFormatProperties(final String valueFormat) {
       builder.put(DelimitedFormat.DELIMITER, delimiter);
     }
 
-    if (ProtobufFormat.NAME.equalsIgnoreCase(valueFormat)) {
+    handleNullableProtobufProperty(valueFormat, builder,
+        CommonCreateConfigs.VALUE_PROTOBUF_NULLABLE_REPRESENTATION);
+
+    final Optional<Integer> valueSchemaId = getValueSchemaId();
+    valueSchemaId.ifPresent(id ->
+        builder.put(ConnectProperties.SCHEMA_ID, String.valueOf(id)));
+
+    return builder.build();
+  }
+
+  private void handleNullableProtobufProperty(final String valueFormat,
+      final Builder<String, String> builder, final String propertyName) {
+    if (ProtobufFormat.NAME.equalsIgnoreCase(valueFormat)
+        || ProtobufNoSRFormat.NAME.equalsIgnoreCase(valueFormat)) {
```

Comment:
Add unit tests for `PROTOBUF_NOSR`, similar to the existing ones for `PROTOBUF`? Also for CreateSourceProperties below.

Reply:
Done